File: app/src/main/java/org/dhis2/usescases/main/program/SyncConflictAdapter.java
package org.dhis2.usescases.main.program;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import org.dhis2.R;
import org.dhis2.databinding.ItemSyncConflictBinding;
import org.hisp.dhis.android.core.imports.TrackerImportConflict;

import java.util.List;

public class SyncConflictAdapter extends RecyclerView.Adapter<SyncConflictHolder> {

    private final List<StatusLogItem> conflicts;

    public SyncConflictAdapter(List<StatusLogItem> conflictList) {
        this.conflicts = conflictList;
    }

    @NonNull
    @Override
    public SyncConflictHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemSyncConflictBinding binding = DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()), R.layout.item_sync_conflict, parent, false);
        return new SyncConflictHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull SyncConflictHolder holder, int position) {
        holder.bind(conflicts.get(position));
    }

    @Override
    public int getItemCount() {
        return conflicts.size();
    }

    public void addItems(List<StatusLogItem> conflicts) {
        this.conflicts.clear();
        this.conflicts.addAll(conflicts);
        notifyDataSetChanged();
    }

    public void addItem(StatusLogItem item){
        this.conflicts.add(item);
        notifyDataSetChanged();
    }

    public void addAllItems(List<StatusLogItem> conflicts){
        this.conflicts.addAll(conflicts);
        notifyDataSetChanged();
    }
}

File: app/src/main/java/org/dhis2/utils/granular_sync/SyncConflictAdapter.kt
package org.dhis2.utils.granular_sync

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView

import org.dhis2.R
import org.dhis2.databinding.ItemSyncConflictBinding

class SyncConflictAdapter (private val conflicts: MutableList<StatusLogItem>) : RecyclerView.Adapter<SyncConflictHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SyncConflictHolder {
        val binding = DataBindingUtil.inflate<ItemSyncConflictBinding>(LayoutInflater.from(parent.context), R.layout.item_sync_conflict, parent, false)
        return SyncConflictHolder(binding)
    }

    override fun onBindViewHolder(holder: SyncConflictHolder, position: Int) {
        holder.bind(conflicts[position])
    }

    override fun getItemCount(): Int {
        return conflicts.size
    }

    fun addItems(conflicts: List<StatusLogItem>) {
        this.conflicts.clear()
        this.conflicts.addAll(conflicts)
        notifyDataSetChanged()
    }

    fun addItem(item: StatusLogItem) {
        this.conflicts.add(item)
        notifyDataSetChanged()
    }

    fun addAllItems(conflicts: List<StatusLogItem>) {
        this.conflicts.addAll(conflicts)
        notifyDataSetChanged()
    }
}
